src/appkit.cs

```diff
@@ -7457,7 +7457,7 @@ interface NSGestureRecognizer : NSCoding {
 		Selector Action { get; set; }
 
 		[Export ("state")]
-		NSGestureRecognizerState State { get; }
+		NSGestureRecognizerState State { get; set; }
```

Comment:
Is it normally settable from obj-c?  If not, should there be an [Advice] or anything on the base version?

Reply:
From the bug:

```
So in obj-c they declare it read only

`@property (readonly) NSGestureRecognizerState state;`

and later use this to all subclases to set
    
    @interface NSGestureRecognizer (NSSubclassUse)
    
    // readonly for users of a gesture recognizer. may only be changed by direct subclasses of NSGestureRecognizer
    @property NSGestureRecognizerState state;  // the current state of the gesture recognizer. can only be set by subclasses of NSGestureRecognizer, but can be read by consumers
    
    
In iOS we handle this correctly:

		[Export ("state")]
		UIGestureRecognizerState State { get; set; }

but do not in XM:

		[Export ("state")]
		NSGestureRecognizerState State { get; }
```

Reply:
I should have added some of it to the commit message. I will on squash.

Reply:
Yah, I got what it was doing from the bug, I was more wondering if the setter in the base class should have something like [Advice ("Only for use in subclasses")] or something like that.  Not sure if you'd ever being using a base NSGestureRecognizer to see the advice, but figured its worth asking.

Reply:
Not a terrible idea.